### PR TITLE
[IOTDB-6251] Make read in SchemaRegion linearizable by default

### PR DESCRIPTION
**Pull request by SzyWilliam**
See https://issues.apache.org/jira/browse/IOTDB-6251
See experiments in https://apache-iotdb.feishu.cn/docx/NJ7vdd9Kfo3nx2xwalZcqBeunr8

### CODE REVIEW

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 1]
```diff
@@ -327,21 +331,25 @@ public DataSet read(ConsensusGroupId groupId, IConsensusRequest request)
...
-        !canServeStaleRead.computeIfAbsent(groupId, id -> new AtomicBoolean(false)).get();
+        readOption == RatisConfig.Read.Option.LINEARIZABLE
+            || !canServeStaleRead.computeIfAbsent(groupId, id -> new AtomicBoolean(false)).get();
 
     RaftClientReply reply;
     try {
       reply = doRead(raftGroupId, request, isLinearizableRead);
       // allow stale read if current linearizable read returns successfully
-      if (isLinearizableRead) {
+      if (readOption == RatisConfig.Read.Option.DEFAULT && isLinearizableRead) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:345]: Optional.ofNullable.IfPresent ?

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 2]
```diff
@@ -766,6 +774,14 @@ private RaftClientReply sendReconfiguration(RaftGroup newGroupConf)
     return reply;
   }
 
+  private void onLeaderChanged(RaftGroupMemberId groupMemberId, RaftPeerId leaderId) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:777]: Optional.ofNullable.IfPresent ?

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/utils/Utils.java [hunk 3]
```diff
@@ -307,11 +307,7 @@ public static void initRatisConfig(RaftProperties properties, RatisConfig config
     RaftServerConfigKeys.Rpc.setFirstElectionTimeoutMax(
         properties, config.getRpc().getFirstElectionTimeoutMax());
 
-    RaftServerConfigKeys.Read.Option option =
-        config.getRead().getReadOption() == RatisConfig.Read.Option.DEFAULT
-            ? RaftServerConfigKeys.Read.Option.DEFAULT
-            : RaftServerConfigKeys.Read.Option.LINEARIZABLE;
-    RaftServerConfigKeys.Read.setOption(properties, option);
+    RaftServerConfigKeys.Read.setOption(properties, RaftServerConfigKeys.Read.Option.LINEARIZABLE);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/utils/Utils.java:310]: use value in the config?

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Meticulous work!
I have left some issues, PTAL

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 1 above)
Comment (SzyWilliam) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java]: Done

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 2 above)
Comment (SzyWilliam) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:777]: Done

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**